sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -974,6 +974,28 @@ def get_cmis_media_lanes_mask(self, api, appl, lport, subport):
 
         return media_lanes_mask
 
+    def reset_app_code_and_reinit_all_lanes(self, api):
+        """
+	   Rest and DP init all the lanes
+        """
+        api.set_application(0xff, 0, 0)
+        api.set_datapath_deinit(0xff)
+        if not api.scs_apply_datapath_init(0xff):
+            return False
+        return True        
+
+    def is_cmis_app_code_reset_required(self, api, app_new):
+        """
+	   Reset app code if non default app code needs to configured 
+        """
+        for lane in range(self.CMIS_MAX_HOST_LANES):
+            app_cur = api.get_application(lane)
+            if app_cur != 0 and app_cur != app_new:
+                self.log_notice("Changing from default AppSel {} to non default AppSel code {}. Reset AppSel "
```

Comment:
@AnoopKamath please move this log to the reset appl function

Reply:
Done